commons-hbase/src/main/scala/com/wajam/commons/hbase/HBaseSerializer.scala

```diff
@@ -29,19 +29,40 @@ class HBaseJsonSerializer(defaultColumnFamily: String = "base",
   val DATE_BYTE = 't'.toByte
   val DATE_STR = DATE_BYTE.toString
 
+  private[hbase] def typeHints: TypeHints = NoTypeHints
+
+  private[hbase] def tablesColumnsFamilies: Map[TableName, Iterable[HBaseSchema.Family]] = Map()
+
   /**
-   * Type hints to support persistence of polymorphic objects.
+   * Serializer with given <code>TypeHint</code>s. Useful to support persistence of polymorphic objects.
    * Example: if we have a Doc and Cat class that inherits Animal, we need to hint that
    * we may use them in a polymorphic way.
-   *
-   * Usage: override def typeHints = ShortTypeHints(List(classOf[Dog], classOf[Cat]))
+   * <p>
+   * Usage: <code>serializer.withTypeHints(ShortTypeHints(List(classOf[Dog], classOf[Cat])))</code>
    */
-  def typeHints: TypeHints = NoTypeHints
+  def withTypeHints(hints: TypeHints): HBaseJsonSerializer = {
+    val families = tablesColumnsFamilies
+    new HBaseJsonSerializer(defaultColumnFamily, extraFormatSerializers) {
```

Comment:
Why not add them as fields to the class instead? It looks a bit awkward and you are probably closing on the parent class everytime you are doing it.